src/main/java/org/jruby/ext/openssl/ASN1.java

```diff
@@ -1395,22 +1398,51 @@ final ASN1TaggedObject toASN1TaggedObject(final ThreadContext context) {
 
             final IRubyObject value = callMethod(context, "value");
             if (value instanceof RubyArray) {
+                // Cruby openssl joins elements of array and casts to string
                 final RubyArray arr = (RubyArray) value;
-                assert ! arr.isEmpty();
 
+                String values = new String();
                 ASN1EncodableVector vec = new ASN1EncodableVector();
+
                 for (final IRubyObject obj : arr.toJavaArray()) {
-                    ASN1Encodable data = ((ASN1Data) obj).toASN1(context);
-                    if ( data == null ) break;
-                    vec.add( data );
+                    if (obj instanceof ASN1Data) {
+                        ASN1Encodable data = ((ASN1Data) obj).toASN1(context);
+                        if (data == null) break;
+                        vec.add(data);
+                    } else {
+                        final IRubyObject string = obj.checkStringType();
+                        if (string instanceof RubyString) {
+                            values = values.concat(string.asJavaString());
```

Comment:
Java usually uses `StringBuilder` for cases like these

Reply:
thx for the suggestion 👍